python/src/numerous/numerous_client.py

```diff
@@ -0,0 +1,70 @@
+"""GraphQL client wrapper for numerous."""
+
+import asyncio
+import os
+from typing import Optional, Union
+
+from numerous.generated.graphql.client import Client
+from numerous.generated.graphql.collection_create import (
+    CollectionCreateCollectionCreateCollection,
+    CollectionCreateCollectionCreateCollectionNotFound,
+)
+from numerous.generated.graphql.fragments import CollectionKey
+
+API_URL_NOT_SET="NUMEROUS_API_URL environment variable is not set"
+MESSAGE_NOT_SET = "NUMEROUS_API_ACCESS_TOKEN environment variable is not set"
+class NumerousClient:
+    def __init__(self, client: Client)->None:
+        self.client = client
+        self.organization_id = ""
+        auth_token = os.getenv("NUMEROUS_API_ACCESS_TOKEN")
+        if not auth_token:
+            raise ValueError(MESSAGE_NOT_SET)
+
+        self.kwargs = {"headers": {"Authorization": f"Bearer {auth_token}"}}
+
+    def _create_collection_key(self,
+                               collection_response:Union[CollectionCreateCollectionCreateCollection,
```

Comment:
Here I think you can use the fragment type `CollectionKey` since `CollectionCreateCollectionCreateCollection` implements it.

Reply:
Also add a fragment for `CollectionNotFound` to make it  even shorter :+1: